README.md

```diff
@@ -30,27 +51,249 @@ npm install --save ipfsd-ctl
+### Remote node
+
+```js
+// Start a remote disposable node, and get access to the api
+// print the node id, and stop the temporary daemon
+
+const DaemonFactory = require('ipfsd-ctl')
+
+const port = 9999
+const server = DaemonFactory.createServer(port)
+const df = DaemonFactory.create({ remote: true, port })
+server.start((err) => {
+  if (err) {
+    throw err
+  }
+
+  df.spawn((err, ipfsd) => {
+    if (err) {
+      throw err
+    }
+
+    ipfsd.api.id(function (err, id) {
+      console.log(id)
+      ipfsd.stop(server.stop)
+    })
+  })
+})
+```
+
+It's also possible to start the server from `.aegir` `pre` and `post` hooks. 
+
+```js
+'use strict'
+
+const createServer = require('./src').createServer
+
+const server = createServer()
+module.exports = {
+  karma: {
+    files: [{
+      pattern: 'test/fixtures/**/*',
+      watched: false,
+      served: true,
+      included: false
+    }],
+    singleRun: true
+  },
+  hooks: {
+    browser: {
+      pre: server.start.bind(server),
+      post: server.stop.bind(server)
+    }
+  }
+}
+```
+
+## Disposable vs non Disposable nodes
+
+`ipfsd-ctl` can create two types of node controllers, `disposable` and `non-disposable`. A disposable node will be created on a temporary repo which will be optionally initialized and started (the default), as well cleaned up on process exit. A non-disposable node on the other hand, requires the user to initialize and start the node, as well as stop and cleanup after wards. Additionally, a non-disposable will allow you to pass a custom repo using the `repoPath` option, if the `repoPath` is not defined, it will use the default repo for the node type (`$HOME/.ipfs` or `$HOME/.jsipfs`). The `repoPath` parameter is ignored for disposable nodes, as there is a risk of deleting a live repo.
+
+## IPFS executable types
+
+`ipfsd-ctl` allows spawning different types of executables, such as:
+
+> `go`
+
+Invoking `df.spawn({type: 'go', exec: '<path to go executable or empty to allow ipfsd to find one automatically>'})` will spawn a `go-ipfs` node.
+
+> `js`
+
+Invoking `df.spawn({type: 'js', exec: '<path to js executable or empty to allow ipfsd to find one automatically>'})` will spawn a `js-ipfs` node.
+
+> `proc`
+
+Invoking `df.spawn({type: 'proc', exec: require('ipfs')})` will spawn an `in-process-ipfs` node using the provided code reference that implements the core IPFS API. Note that, `exec` is required if `type: 'proc'` is used.
+
+### IPFS executables
+
+`ipfsd-ctl` no longer installs go-ipfs nor js-ipfs dependencies, instead it expects them to be provided by the parent project. In order to be able to use both go and js daemons, please make sure that your project includes these two npm packages as dependencies.
+
+- `ipfs` - the js-ipfs implementation
+- `go-ipfs-dep` - the packaged go-ipfs implementation
+
+## API
+
+### Daemon Factory
+
+#### Create a `DaemonFactory`
```

Comment:
Put in the title the call: "Create a `DaemonFactory` - `const df = DaemonFactory.create([options])`